pkg/resources/validation/cluster/provisioningcluster.go

```diff
@@ -135,6 +147,36 @@ func validateCreatorID(request *webhook.Request, oldCluster, cluster *v1.Cluster
 	return nil
 }
 
+func (p *provisioningClusterValidator) validateClusterName(request *webhook.Request, response *webhook.Response, cluster *v1.Cluster) error {
+	if request.Operation != admissionv1.Create {
+		return nil
+	}
+
+	// Look for an existing management cluster with the same name. If a management cluster with the given name does not
+	// exists, then it should be checked that the provisioning cluster the user is trying to create is not of the form
+	// "c-xxxxx" because names of that form are reserved for "legacy" management clusters.
+	_, err := p.mgmtClusterClient.Get(cluster.Name, metav1.GetOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
+
+	if cluster.Name == "local" && cluster.Namespace != "fleet-local" ||
+		mgmtNameRegex.MatchString(cluster.Name) && err != nil {
+		// A provisioning cluster with name "local" is only expected to be created in the "fleet-local" namespace.
+		// Also, a provisioning cluster with name of the form "c-xxxxx" is expected to be created if a management cluster
+		// of the same name already exists because Rancher will create such a provisioning cluster.
+		// Any other situation should not be allowed.
+		response.Result = &metav1.Status{
+			Status:  "Failure",
+			Message: "cluster name cannot be \"local\" nor of the form \"c-xxxxx\"",
+			Reason:  metav1.StatusReasonInvalid,
+			Code:    http.StatusUnprocessableEntity,
+		}
+	}
```

Comment:
Do you think you could put this in its own function and write a few unit test case for it?

Reply:
@rmweir , generally I would be all for doing this, but on what is essentially the last day before "code freeze" I think it's not the best time to start enforcing more unit tests. :) @thedadams , unless you have some extra bandwidth to get it in.

Reply:
I can put some in tomorrow.

Reply:
Unit test has been added.

Reply:
Thanks for adding the test, it looks good.